### PR TITLE
Refactored back-end for version/spec independence

### PR DESCRIPTION
**Pull request by pl9ed**
Uses intermediary Docutest, Request, and Endpoint objects.

### CODE REVIEW

File: src/main/java/com/revature/models/SwaggerDocutest.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import com.revature.services.OASService;
+import io.swagger.models.Swagger;
+import lombok.Data;
+
+@Data
+public class SwaggerDocutest implements Docutest {
+    int id; // id of swaggersummary object
+    private List<Request> requests;
+    
+    // TODO autowire once Spring breans are set up
```
Comment (IkenoXamos) [inline src/main/java/com/revature/models/SwaggerDocutest.java]: Spring Beans are as easy to set up as adding corresponding Stereotype annotations, so if this needs to be autowired, go ahead and setup the beans that are needed. Note that this class would additionally need to be a Spring Bean in order to have its dependencies be autowired.

File: src/main/java/com/revature/services/OASService.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+                    req.setVerb(operationEntry.getKey());
+                    
+                    
+                    // doesn't seem to be any requirements for indices, so we need to check if parameter
+                    // is an instance of BodyParam/HeaderParam/etc
+                    List<Parameter> params = operationEntry.getValue().getParameters();
+                    
+                    for (Parameter param : params) {
+                        if (param instanceof BodyParameter) {
+                            // TODO set body params
```
Comment (IkenoXamos) [inline src/main/java/com/revature/services/OASService.java]: I imagine that this feature is intended to be constructed in a separate PR?

**Review by IkenoXamos — CHANGES_REQUESTED** [review]

**Comment by IkenoXamos** [comment]
Can you also review some of the code smells regarding this PR?

File: src/main/java/com/revature/services/OASService.java (hunk 2 above)
Comment (pl9ed) [reply src/main/java/com/revature/services/OASService.java]: Yup. It's currently being worked on in the createData branch, but since we refactored the backend, I'll probably want to merge this with master and rebase before continuing work on it.

File: src/main/java/com/revature/models/SwaggerDocutest.java (hunk 1 above)
Comment (pl9ed) [reply src/main/java/com/revature/models/SwaggerDocutest.java]: I decided to not inject the service, since it was causing more problems than it was worth. I wanted to have a constructor that takes in a Swagger object as an input, but then Spring would need that Swagger object to also be a Spring bean, which isn't what we want. 

If I use either constructor or setter injection for the OASService, the constructor that takes in the Swagger object does not have the service injected (i.e. adapter is null). Field injection with @Autowired doesn't seem to work either, although I'm not sure why. I originally thought it was because Spring was using that constructor, but having a separate annotated constructor ended up producing the same result. 

None of the following work:
```Java
private OASService adapter;

public Docutest(Swagger input) {
...
}

@Autowired
public setAdapter(OASService adapter) {
...
}
```

```Java
@Autowired
public Docutest(OASService adapter) { 
... 
}

public Docutest(Swagger input) {
...
}
```

```Java
@Autowired
private OASService adapter

public Docutest(Swagger input) {
...
}
```

```Java
@Autowired
private OASService adapter

@Autowired
public Docutest(OASService adapter) {
...
}
public Docutest(Swagger input) {
...
}
```

**Comment by pl9ed** [comment]
@IkenoXamos I removed an unused declaration in the Controller and put some of the functionality in a helper method, but because of the way the information is structured in a series of nested HashTrees in the Swagger object, I'd have to pass a ton of information to completely remove nested for loops/if statements, and I feel that actually makes the code less readable.

The rest of the code smells are TODOs/unused declarations that are essentially placeholders for stuff that's currently being worked on in other branches.

File: src/main/java/com/revature/services/OASService.java (hunk 2 above)
Comment (IkenoXamos) [reply src/main/java/com/revature/services/OASService.java]: Sounds good

File: src/main/java/com/revature/models/SwaggerDocutest.java (hunk 1 above)
Comment (IkenoXamos) [reply src/main/java/com/revature/models/SwaggerDocutest.java]: I would recommend removing the custom constructor, and allowing Lombok to generate a Constructor that accepts just the list of requests (or we can create that constructor ourselves if Lombok is inconvenient). Then externally, we can create logic that will use the OASService to obtain the requests. This will allow the model class to not include any logic, which means it can be safely excluded from analysis reports. And the logic for obtaining the requests can be tested and written independently.

**Review by IkenoXamos — CHANGES_REQUESTED** [review]
Please refactor according to the comment I left in response.

File: src/test/java/com/revature/services/JMeterServiceTest.java [hunk 3]
```diff
@@ -20,23 +19,25 @@
...
-
 class JMeterServiceTest {
 
     private JMeterService jm;
     private LoadTestConfig loadConfig = new LoadTestConfig();
     private static final String JMeterPropPath = "src/test/resources/test.properties";
     private static final String CSV_FILE_PATH = "./datafiles/user_0.csv";
-    public static final String DIRECTORY_PATH = "./datafiles";
+    private static final String DIRECTORY_PATH = "./datafiles";
+    private static OASService adapter = new OASService();
```
Comment (IkenoXamos) [inline src/test/java/com/revature/services/JMeterServiceTest.java:34]: So far we still are unable to use `@Autowired` annotations within tests?

**Review by IkenoXamos — COMMENTED** [review]
Quick question

File: src/test/java/com/revature/services/JMeterServiceTest.java (hunk 3 above)
Comment (pl9ed) [reply src/test/java/com/revature/services/JMeterServiceTest.java:34]: SpringBootTest runs a lot slower since it needs to set up all of our beans and dependencies. Since these are unit tests, I just set up the service as a static field. I was planning on having a separate class for integration tests using the webapp context.

Comment (IkenoXamos) [reply src/test/java/com/revature/services/JMeterServiceTest.java:34]: Alrighty, that sounds good.

**Review by IkenoXamos — APPROVED** [review]
Looks good
